### PR TITLE
DATA: Agreement Model & Migration

### PR DESCRIPTION
**Pull request by NaynaMallick**

### CODE REVIEW

File: SCMS.Services.Api/Migrations/20220222181518_InitializeModels.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 namespace SCMS.Services.Api.Migrations
 {
-    public partial class ReInitializeModels : Migration
+    public partial class InitializeModels : Migration
```
Comment (ShriHumrudha) [inline SCMS.Services.Api/Migrations/20220222181518_InitializeModels.cs:12]: You need to delete the Migrations folder every time and re-create.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: SCMS.Services.Api/Migrations/20220222181518_InitializeModels.cs (hunk 1 above)
Comment (NaynaMallick) [reply SCMS.Services.Api/Migrations/20220222181518_InitializeModels.cs]: done

File: SCMS.Services.Api/Brokers/Storages/StorageBroker.Agreement.References.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using Microsoft.EntityFrameworkCore;
+using SCMS.Services.Api.Models.Foundations.Agreements;
+
+namespace SCMS.Services.Api.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+        public static void SetAgreementReferences(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Agreement>()
```
[the remark this answers is not in the document]
Comment (NaynaMallick) [reply SCMS.Services.Api/Brokers/Storages/StorageBroker.Agreement.References.cs:14]: added

File: SCMS.Services.Api/Brokers/Storages/StorageBroker.Agreement.References.cs [hunk 3]
```diff
@@ -22,6 +22,12 @@ public static void SetAgreementReferences(ModelBuilder modelBuilder)
                 .WithMany(user => user.UpdatedAgreements)
                 .HasForeignKey(agreement => agreement.UpdatedBy)
                 .OnDelete(DeleteBehavior.NoAction);
+
+            modelBuilder.Entity<Agreement>()
+                .HasOne(agreement => agreement.TermsAndCondition)
+                .WithMany(TermsAndCondition => TermsAndCondition.TermsAndConditiondAgreements)
```
Comment (ShriHumrudha) [inline SCMS.Services.Api/Brokers/Storages/StorageBroker.Agreement.References.cs:28]: small t

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]
Do you need to run the migration with this change?

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**